### PR TITLE
leave out part of cross validation

### PR DESCRIPTION
**Pull request by Farnazmdi**
1. The estimators are updated such that they ignore negative observations.
2. During the calculation of pdf of observations, those cells with negative observation will get an equal probability to each state, this way they do not contribute to fitting.

### CODE REVIEW

File: lineage/states/StateDistributionGamma.py [hunk 1]
```diff
@@ -68,6 +68,8 @@ def pdf(self, x: np.ndarray):
         # Update for observed Bernoulli
         ll[np.isfinite(x[:, 0])] += sp.bernoulli.logpmf(x[np.isfinite(x[:, 0]), 0], self.params[0])
 
+        ll[x[:, 0] == -1] = np.log(1/num_states)
```
Comment (aarmey) [inline lineage/states/StateDistributionGamma.py:71]: Why do you need this? It seems like you should be able to just set the log-likelihood to 0.

File: lineage/states/stateCommon.py [hunk 2]
```diff
@@ -79,20 +79,25 @@ def gamma_estimator(gamma_obs: list[np.ndarray], time_cen: list[np.ndarray], gam
...
 
-    if len(gamma_obs) == 4:  # for constrained optimization
+    if len(gamma_obs_) == 4:  # for constrained optimization
         A = np.zeros((3, 5))  # is a matrix that contains the constraints. the number of rows shows the number of linear constraints.
         np.fill_diagonal(A[:, 1:], -1.0)
         np.fill_diagonal(A[:, 2:], 1.0)
         linc = [LinearConstraint(A, lb=np.zeros(3), ub=np.full(3, np.inf))]
         if np.allclose(np.dot(A, x0), 0.0):
-            x0 = np.array([200.0, 0.2, 0.4, 0.6, 0.8])
+            x0 = np.array([200.0, 0.5, 0.2, 0.4, 0.6, 0.8])
```
Comment (aarmey) [inline lineage/states/stateCommon.py:100]: Why is there a new parameter here? Or has it been wrong?

**Review by aarmey — CHANGES_REQUESTED** [review]
Generally good. Just a couple questions.

File: lineage/states/StateDistributionGamma.py (hunk 1 above)
Comment (Farnazmdi) [reply lineage/states/StateDistributionGamma.py:71]: It was my first intuition, but putting the likelihood of those cells to zero will create some problems in the later steps that something will be divided by this value and threw an error.

File: lineage/states/stateCommon.py (hunk 2 above)
Comment (Farnazmdi) [reply lineage/states/stateCommon.py:100]: oops, my bad. it was right, I will fix it now.

File: lineage/states/StateDistributionGamma.py (hunk 1 above)
Comment (Farnazmdi) [reply lineage/states/StateDistributionGamma.py:71]: I have not tried setting it to a small number like epsilon though. I will try that and see if it still errors.

Comment (aarmey) [reply lineage/states/StateDistributionGamma.py:71]: This is more reason to set it to 0! Under no circumstances should the code be dividing by the log-likelihood. If a 0 causes an error here, that indicates a problem that needs to be fixed elsewhere.

Comment (Farnazmdi) [reply lineage/states/StateDistributionGamma.py]: I think your recent changes to the estimators fixed that. I don't get that error anymore when I set the log-likelihood to zero, but get more cost function = inf.
I am trying to find a way to make the `minimize` run again if the cost function becomes inf.

Comment (aarmey) [reply lineage/states/StateDistributionGamma.py]: Hmm... where does inf first arise? Is it in the estimator? It's likely a quick fix if we can track it down.

Comment (Farnazmdi) [reply lineage/states/StateDistributionGamma.py]: It is inside the `gamma_LL()` line 51 stateCommon.py (where you have `if np.isinf(outt): print(x)`). The assersion about `outt` being finite is violated.

File: lineage/states/stateCommon.py [hunk 3]
```diff
@@ -79,14 +79,19 @@ def gamma_estimator(gamma_obs: list[np.ndarray], time_cen: list[np.ndarray], gam
     if np.sum([np.sum(g) for g in gammas]) < 0.1:
         gammas = [np.ones(g.size) for g in gammas]
 
+    # make sure all negative observations are removed
+    gamma_obs_ = [gm[gamma_obs[i] >= 0] for i, gm in enumerate(gamma_obs)]
```
Comment (aarmey) [inline lineage/states/stateCommon.py:83]: Does the estimator need to do this step, if it's already done in the state distribution class? It seems like that can just pass the filtered variables...

**Review by aarmey — CHANGES_REQUESTED** [review]

File: lineage/states/stateCommon.py (hunk 3 above)
Comment (Farnazmdi) [reply lineage/states/stateCommon.py]: done.

**Review by aarmey — APPROVED** [review]
Great. Merging!

**Pull request merged**